tech/firewall-rule-format.md

```diff
@@ -0,0 +1,574 @@
+        "name": "default-allow-rdp",
+        "description": "Allow RDP from anywhere",
+        "network": "https://www.googleapis.com/compute/v1/projects/strong-augury-224506/global/networks/default",
+        "priority": 65534,
+        "sourceRanges": [
+          "0.0.0.0/0"
+        ],
+        "allowed": [
+          {
+            "IPProtocol": "tcp",
+            "ports": [
+              "3389"
+            ]
+          }
+        ],
+        "direction": "INGRESS",
+        "logConfig": {
+          "enable": false
+        },
+        "disabled": false,
+        "selfLink": "https://www.googleapis.com/compute/v1/projects/strong-augury-224506/global/firewalls/default-allow-rdp"
+      },
+      {
+        "kind": "compute#firewall",
+        "id": "6979850877904078814",
+        "creationTimestamp": "2018-12-04T04:55:13.187-08:00",
+        "name": "default-allow-ssh",
+        "description": "Allow SSH from anywhere",
+        "network": "https://www.googleapis.com/compute/v1/projects/strong-augury-224506/global/networks/default",
+        "priority": 65534,
+        "sourceRanges": [
+          "0.0.0.0/0"
+        ],
+        "allowed": [
+          {
+            "IPProtocol": "tcp",
+            "ports": [
+              "22"
+            ]
+          }
+        ],
+        "direction": "INGRESS",
+        "logConfig": {
+          "enable": false
+        },
+        "disabled": false,
+        "selfLink": "https://www.googleapis.com/compute/v1/projects/strong-augury-224506/global/firewalls/default-allow-ssh"
+      }
+    ]
+
+Since GCP returns one JSON object per firewall rule while Azure groups
+firewall rules under a network security group, the Azure cloud plugin
+should generate multiple firewall rule records, one for each firewall
+rule in a network security group in order to maintain parity with the
+records of GCP firewall rules.
+
+Further the `com` (common notation) bucket of the generated records
+should be populated with generic attributes that are common in both
+Azure and GCP firewall rules.
+
+From the next section onwards, we will compare a single firewall rule,
+i.e., a single item in the `security_rules` bucket in Azure with a
+single `compute#firewall` document in GCP for various types of firewall
+rules.
+
+
+Multiple Ranges
+---------------
+
+There are primarily three important pieces of information we need to
+analyze in a firewall rule to understand if the firewall rule is
+potentially weak:
+
+  - Direction: We will focus on inbound rules first. Azure calls it
+    `Inbound`. GCP calls it `INGRESS`. In the common notation, we will
+    write it as `in`.
+
+  - Access: Rules can either deny access or allow access. We will focus
+    only on those that allow access.
+
+  - Source address range: We need to understand how they look in the
+    JSON data when multiple ranges are defined. We want to find those
+    rules that have `*` or `0.0.0.0/0` or any other pattern that
```

Comment:
Not sure why github is rendering everything after this line in the commit diff in blue color 😅 . It seems like the syntax for inline code and italicized font is getting mixed up. The actual markdown rendering [here](https://github.com/cloudmarker/guides/blob/e5ee0c2d47ea7f172292f7ad47bf73b30ad72a48/tech/firewall-rule-format.md#multiple-ranges) seems to be correct.

Reply:
Interesting observation. This looks like a bug in the syntax highlighter for diff view.